### PR TITLE
KFSPTS-22328 create concur event notification v2 results report

### PR DESCRIPTION
**Pull request by jhulslander**
this is NOT ready for review yet

### CODE REVIEW

**Review by db483 — CHANGES_REQUESTED** [review]
Changes look good but I had some small suggestions for improvement.

File: src/main/resources/CU-ApplicationResources.properties [hunk 1]
```diff
@@ -715,3 +715,12 @@ message.one.to.one.iso.to.fips.mappping=One Active ISO-to-FIPS Country generic m
...
 error.many.fips.to.iso.mappings=More than one Active FIPS-to-ISO Country generic mapping found for FIPS Country code : {0}
 message.one.to.one.fips.to.iso.mapping=One Active FIPS-to-ISO Country generic mapping found FIPS Country code : {0} mapped to ISO Country code : {1}
+
+concur.event.v2.processing.report.section.opening=****************************** {0} ****************************************
+concur.event.v2.processing.report.section.closing=***************************************************************************
+concur.event.v2.processing.report.summary.line={0} with {1}: {2}
+concur.event.v2.processing.report.detail.summary={0} with {1}
+concur.event.v2.processing.report.detail.item={0}: {1}
+concur.event.v2.processing.report.title=Concur Event Notification V2. Processing Report
+concur.event.v2.processing.report.file.name=Concurn_Event_Notification_V2_Processing_Results_
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/resources/CU-ApplicationResources.properties:725]: ok

File: src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurEventNotificationV2ReportServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        List<String> toAddressList = new ArrayList<>();
+        toAddressList.add(toAddress);
+        
+        BodyMailMessage message = new BodyMailMessage();
+        message.setFromAddress(fromAddress);
+        message.setSubject(subject);
+        message.getToAddresses().addAll(toAddressList);
+        message.setMessage(body);
+        
+        boolean htmlMessage = false;
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurEventNotificationV2ReportServiceImpl.java:146]: I will do the direct method

File: src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurEventNotificationV2ReportServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        return contents;
+    }
+    
+    public void sendEmail(String subject, String body) {
+        String toAddress = concurBatchUtilityService.getConcurParameterValue(ConcurParameterConstants.CONCUR_REPORT_EMAIL_TO_ADDRESS);
+        String fromAddress = concurBatchUtilityService.getConcurParameterValue(ConcurParameterConstants.CONCUR_REPORT_EMAIL_FROM_ADDRESS);
+        List<String> toAddressList = new ArrayList<>();
+        toAddressList.add(toAddress);
+        
+        BodyMailMessage message = new BodyMailMessage();
```
[the remark this answers is not in the document]
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurEventNotificationV2ReportServiceImpl.java:140]: sure

File: src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurEventNotificationV2ReportServiceImpl.java [hunk 4]
```diff
@@ -132,6 +132,22 @@ protected String readReportFileToString(File reportFile) {
     }
     
     public void sendEmail(String subject, String body) {
+        BodyMailMessage message = buildBodyMailMessage(subject, body);
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("sendEmail, from address: " + message.getFromAddress());
+            LOG.debug("sendEmail. Send email to: " + String.join(",", message.getToAddresses()));
+            LOG.debug("sendEmail, the email subject: " + subject);
+            LOG.debug("sendEmail, the email budy: " + body);
```
Comment (db483) [inline src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurEventNotificationV2ReportServiceImpl.java:140]: I think this should be "the email body"

**Comment by db483** [comment]
Changes look good. There is just one misspelling to be corrected. Thank you.

File: src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurEventNotificationV2ReportServiceImpl.java (hunk 4 above)
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurEventNotificationV2ReportServiceImpl.java:140]: ok

**Comment by jhulslander** [comment]
I fixed the typo .... please hold off on merging this PR though.  I am waiting for Tami to create some invalid account requests in Concur and for Tami to give feedback.

**Comment by db483** [comment]
Sounds good. Thank you.

**Comment by jhulslander** [comment]
I have made some changes based on feedback from Tami and Cathy.  This is ready for review again

**Review by db483 — APPROVED** [review]
Changes look good.

**Pull request merged**